src/providers/http.ts

```diff
@@ -43,14 +53,17 @@ export class HttpProvider implements ApiProvider {
     // Render all nested strings
     const nunjucks = getNunjucksEngine();
     const stringifiedConfig = safeJsonStringify(this.config);
-    const renderedConfig: { method: string; headers: Record<string, string>; body: any } =
-      JSON.parse(
-        nunjucks.renderString(stringifiedConfig, {
-          prompt,
-          ...context?.vars,
-        }),
-      );
-
+    const renderedConfigString = nunjucks.renderString(stringifiedConfig, {
+      // nunjucks does not escape JSON strings, so we need to escape them manually
+      prompt: isValidJson(prompt) ? prompt.replace(/"/g, '\\"') : prompt,
```

Comment:
If it is valid JSON then we are escaping the strings. but if it is not valid, then we are using `prompt` as it is..?

Reply:
Thanks for the quick comment. Good question. Some prompts are defined like: 

```yaml
prompts:
  - Tell me about {{topic}}
```
This is designed to cover both scenarios. I'll make sure it's covered in the test cases for you. If there's something I am missing please let me know and I'll attempt to fix it.

Reply:
Note that in order to get this format to work you have to modify the messages parameter

```yaml
description: 'LM Studio Eval Example'
prompts:
  - Tell me about {{topic}}
providers:
  - id: 'http://localhost:1234/v1/chat/completions'
    config:
      method: 'POST'
      headers:
        'Content-Type': 'application/json'
      body:
        messages: '[{"role": "user", "content": "{{ prompt }}"}]'
        model: 'bartowski/gemma-2-9b-it-GGUF'
        temperature: 0.7
        max_tokens: -1
      responseParser: 'json.choices[0].message.content'
tests:
  - vars:
      topic: bananas
```